scaffolds/Magic Auth With NextJS/template/src/pages/index.tsx

```diff
@@ -31,7 +32,7 @@ export default function Home() {
         <div className="w-[100%] grid grid-cols-6 gap-2 p-4">
           {authMethods.map((method) =>
             React.createElement(
-              lazy(() => import('../components/auth/' + method.replaceAll(' ', ''))),
+              dynamic(() => import('../components/auth/' + method.replaceAll(' ', '')), { ssr: false }),
```

Comment:
@AdityaKulkarni is there a way for us to get this in the imports at the top of file? Like maybe add all the imports with the templating syntax <%=...%> and then inside that have it evaluate to an empty string or something if it isn't one they chose? Then something similar down on line 35. Like include all the auth methods each as a separate template syntax where unchosen ones evaluate to nothing and the others evaluate to the right component name?

Reply:
Just thinking through options here for making it a nicer developer experience after it's cloned. This absolutely works but is tougher to grok than the traditional import and usage syntax

Reply:
What I can do is, get that import out of the render function and put it in a function at the top of the file with the <%=..%> template. It would look something like
```typescript
const getSelectedAuthComponents = () => {
	const authMethods: string[] = [<%- selectedAuthTypes.map(authType => `"${authType}"`) %>]
	const importedComponents: any[] = []
	authMethods.map(method => importedComponents.push(import('../components/auth/' + method.replaceAll(' ', ''))))
	return importedComponents
}

```